frontend/src/views/admin/matching/match-actions/modify-match-status.tsx

```diff
@@ -0,0 +1,109 @@
+import React from "react";
+import { useThunkDispatch } from "../../../../libs/thunk-dispatch";
+import { upsertMatch } from "../actions";
+import { ApplicantSummary, MatchableAssignment } from "../types";
+
+/**
+ * Update a match to mark an applicant as being staged-assigned to a
+ * position, with `hoursAssigned` number of hours.
+ */
+export function AssignApplicantToPosition(
```

Comment:
You're writing a bunch of custom hooks. Hooks should start with `use`, so, e.g., this should be called `useAssignApplicantToPosition`. The same for all the others.

When hooks are used, they should be assigned to constants first-thing in your components before any possible conditional logic.